src/cookie.rs

```diff
@@ -0,0 +1,252 @@
+// Based on https://github.com/square/okhttp/blob/master/okhttp/src/main/java/okhttp3/Cookie.java
+// and http://doc.qt.io/qt-5/qnetworkcookie.html.
+
+use std::time::Duration;
+
+/// Represents an HTTP cookie. It has a name, a single value and optional parameters to
+/// maintain persistent information across requests.
+///
+/// Check out the [official specification](https://tools.ietf.org/html/rfc6265).
+#[derive(Clone, Debug)]
+pub struct Cookie<S> {
+    pub(self) domain: Option<S>,
+    pub(self) expires_at: Option<Duration>,
+    pub(self) host_only: bool,
+    pub(self) http_only: bool,
+    pub(self) name: S,
+    pub(self) path: Option<S>,
+    pub(self) persistent: bool,
+    pub(self) secure: bool,
+    pub(self) value: S,
+}
+
+impl<S> Cookie<S>
+where
+    S: AsRef<str>,
+{
+    /// If any, returns the domain associated with this
+    /// cookie. E.g.: `Set-Cookie: name=value; Domain=.foo.com`.
+    pub fn domain(&self) -> Option<&str> {
+        self.domain.as_ref().map(|x| x.as_ref())
+    }
+
+    /// If any, returns the time this cookie expires.
+    ///
+    /// A value less than the current time means that the cookie has been expired.
```

Comment:
Perhaps we should clarify that this is a duration since the unix epoch. It took a bit of browsing to figure out *which* duration we were talking about, haha.

Reply:
Alternatively this API could perhaps expose the value as a `SystemTime` instance so that expiration dates become non-relative, and possibly easier to work with.

The tradeoff might be that the API no longer matches the cookie API 1:1.